dra-evolution/pkg/api/claim_types.go

```diff
@@ -1,379 +1,467 @@
+type StructuredDriverData struct {
+	// Config contains all the configuration pieces that apply to the entire claim
+	// and that were meant for the driver which handles these resources.
+	// They get collected during the allocation and stored here
+	// to ensure that they remain available while the claim is allocated.
 	//
-	// This field is owned by the scheduler, whereas the DeviceStatuses
-	// field is owned by the drivers.
+	// Entries are sorted from "most specific" first to "least specific" last:
+	// - claim
+	// - claim class reference
 	//
 	// +optional
-	// +listType=atomic
-	Allocations []DeviceAllocation `json:"allocations,omitempty"`
+	Config []DriverConfiguration `json:"config,omitempty"`
 
-	// DeviceStatuses contains the status of each device allocated for this
-	// claim, as reported by the driver. This can include driver-specific
-	// information. Entries are owned by their respective drivers.
+	// NodeName is the name of the node providing the necessary resources
+	// if the resources are local to a node.
 	//
 	// +optional
-	// +listType=map
-	// +listMapKey=devicePoolName
-	// +listMapKey=deviceName
-	DeviceStatuses []AllocatedDeviceStatus `json:"deviceStatuses,omitempty"`
+	NodeName string `json:"nodeName,omitempty" protobuf:"bytes,3,name=nodeName"`
 
-	// PodNames contains the names of all Pods using this claim.
-	// TODO: Can we just use ownerRefs instead?
+	// Results lists all allocated driver resources.
 	//
-	// +optional
-	// +listType=set
-	PodNames []string `json:"podNames,omitempty"`
+	// +listType=atomic
+	Results []RequestAllocationResult `json:"results" protobuf:"bytes,4,name=results"`
 }
 
-// NOTE: We no longer have DeviceClaimTemplate. Instead, the PodSpec will
-// directly contain a either a DeviceClaimName (to enable multiple pods to
-// refer to a pre-provisioned claim), or an embedded struct that includes
-// ObjectMeta and a DeviceClaimName. In this case, the named DeviceClaim will
-// be treated as a template; that is, its spec will be copied to create a new
-// claim, based on the new metadata. Re-using claim-as-a-template avoids
-// another, nearly identical top-level API object. But it may be confusing, we
-// need feedback.
-//
-// DeviceClassConfigReference is used to refer to arbitrary configuration
-// objects from the class. Since it is the class, and therefore is created by
-// the administrator, it allows referencing objects in any namespace.
-
-type DeviceClassConfigReference struct {
-	// API version of the referent.
-	// +required
-	APIVersion string `json:"apiVersion"`
-
-	// Kind of the referent.
-	// +required
-	Kind string `json:"kind"`
-
-	// Namespace of the referent.
-	// +required
-	Namespace string `json:"namespace"`
-
-	// Name of the referent.
-	// +required
-	Name string `json:"name"`
-}
+// DriverConfiguration is one entry in a list of configuration pieces.
+type DriverConfiguration struct {
+	// Admins is true if the source of the piece was a class and thus
```

Comment:
Not for `StructuredDriverData.Config` (doesn't inherit from class).

But for `RequestAllocationResult,Config` it still makes sense (inherits from DeviceClass).

I have to fix `StructuredDriverData.Config` so that it doesn't use this type here.